EIPS/eip-6228.md

```diff
@@ -0,0 +1,233 @@
+- Meta-transactions must become standard procedure when interacting with smart contracts.
+- Gas efficiency is in no way less important when it comes to optimistic and zero knowledge roll-ups. If anything, lower network fees enable more automation, which is about to become essential in such a hyper-competitive industry.
+- Full automation can easily be too costly for project funding. Therefore, meta-transactions fees should be deducted directly from the transfer amount.
+
+An example:
+Alice has never had a wallet before but is eligible for an airdrop on Arbitrum. All she needs to do is install a wallet, sign the data, and then she has airdrop tokens minus the keeper fee. She does not have to make any transactions or possess any ETH. She is not limited in any way more than Bob, who is transacting on his own. Alice can vote, stake, claim rewards, trade on Uniswap. In fact, she can do more than Bob, such as subscribing to staking rewards to get them automatically every month, a part of these rewards then automatically sent to pay her bills and support her favorite content creators.
+
+## Motivation
+
+To solve the issues with existing keeper/relayer solutions today:
+
+- Conservative approaches massively limit the functionality, resulting in a lack of significant impact and usage. This EIP tries to acknowledge and embrace the “wild west” of DeFi without introducing more rules than are absolutely necessary.
+- All existing solutions rely on funding/supporting a relayer/keeper network, which makes innovation more dependent on external funding.
+
+## Specification
+
+Signatories can create executable signatures for transactions of any kind. As long as signatory `address(this)` token balance or `returnToken`(token returned after after execution) balance is higher than keeper fee of a transaction at hand, and as long as remaining value after keeper fee transaction is above `minOut`, it is executable.
+
+The returned funds are split between the signatory and the keeper, according to ETH ratio obtained through an on-chain oracle such as Uniswap (which is admittably a bit extreme depending on the case). The same flow applies for ETH as the returned token, minus the oracle price conversion.
+
+Additional options for transactions should be provided to claim staking rewards or an airdrop with the signature even with both ETH and `address(this)` balances at 0. To enable this functionality on a larger scale, functions outside the core token contract which are intended to transfer tokens to the user never actually do so. Instead, to allow the subtraction of the keeper fee in the most efficient way, these functions return the amount (of staking reward to claim, etc.) to the core token contract, where the transfer is executed with the applied fee. This functionality requires treating keeper calls differently, either with an `if` statement or by adding a second version of the call and guarding it with `require` statement.
```

Comment:
You should move anything not explicitly specified in this EIP into its own proposal.